File: 卡码/代码随想录/1/10-单调栈/Go/LeetCode_0739.go
// 739. 每日温度
package main

func dailyTemperatures(temperatures []int) []int {
	res := make([]int, len(temperatures))
	st := make([]int, 0)
	for i := 0; i < len(temperatures); i++ {
		for len(st) != 0 && temperatures[i] > temperatures[st[len(st)-1]] {
			res[st[len(st)-1]] = i - st[len(st)-1]
			st = st[:len(st)-1]
		}
		st = append(st, i)
	}
	return res
}

File: 卡码/代码随想录/1/10-单调栈/Python/LeetCode_0739.py
#739. 每日温度
from typing import *

class Solution:
	def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
		res:List[int]=[0]*len(temperatures)
		st:List[int]=[]
		for i in range(len(temperatures)):
			while len(st) and temperatures[i]>temperatures[st[-1]]:
				res[st[-1]]=i-st[-1]
				st.pop()
			st.append(i)
		return res